src/main/java/step2/domain/entity/Lotto.java

```diff
@@ -57,26 +31,26 @@ public List<LottoNumber> getLottoNumbers() {
         return lottoNumbers;
     }
 
-    public LottoPrize prize(Lotto compareLotto, LottoNumber bonus) {
-        // compare: 로또 번호, lottoNumbers: 당첨 번호
-        final var compare = compareLotto.getLottoNumbers();
-        final var count = (int) lottoNumbers.stream()
-                .filter(compare::contains)
-                .count();
+    @Override
+    public String toString() {
+        return lottoNumbers.toString();
+    }
 
-        if (isSecondBonusLotto(bonus, compare, count)) {
-            return LottoPrize.SECOND_BONUS;
-        }
+    public int compareWinnerLottoNumbers(List<LottoNumber> winnerLottoNumbers) {
+        return (int) lottoNumbers.stream()
+                .filter(winnerLottoNumbers::contains)
+                .count();
+    }
 
-        return LottoPrize.prize(count);
+    public boolean isSecondBonusLotto(int count, LottoNumber bonusNumber) {
+        return lottoNumbers.contains(bonusNumber) && count == SECOND_LOTTO_COUNT;
     }
 
-    private static boolean isSecondBonusLotto(LottoNumber bonus, List<LottoNumber> compare, int count) {
-        return count == SECOND_LOTTO_COUNT && compare.contains(bonus);
+    public boolean isManualLotto() {
+        return this.lottoType == LottoType.MANUAL;
     }
 
-    @Override
-    public String toString() {
-        return lottoNumbers.toString();
+    public boolean isAutoLotto() {
+        return this.lottoType == LottoType.AUTO;
```

Comment:
메시지를 보내는 방식으로 변경해보면 어떨까요?
```suggestion
        return this.lottoType.isAuto();
```